rest/server_context.go

```diff
@@ -1147,6 +1148,70 @@ func (sc *ServerContext) updateCalculatedStats() {
 
 }
 
+func initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPath, caCertPath string) (*gocbcore.Agent, error) {
+	authenticator, err := base.GoCBCoreAuthConfig(clusterUser, clusterPass, certPath, keyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsRootCAProvider, err := base.GoCBCoreTLSRootCAProvider(caCertPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config := gocbcore.AgentConfig{
+		Auth:              authenticator,
+		TLSRootCAProvider: tlsRootCAProvider,
+	}
+
+	err = config.FromConnStr(clusterAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	agent, err := gocbcore.CreateAgent(&config)
+	if err != nil {
+		return nil, err
+	}
+
+	agentReadyErr := make(chan error)
+	_, err = agent.WaitUntilReady(time.Now().Add(15*time.Second), gocbcore.WaitUntilReadyOptions{ServiceTypes: []gocbcore.ServiceType{gocbcore.MgmtService}}, func(result *gocbcore.WaitUntilReadyResult, err error) {
```

Comment:
Have you tested this against a multi-node server, to validate that when clusterAddress is just a single node, you're getting the full set of endpoints back (i.e. ensure agent.WaitUntilReady is actually waiting on the cluster map to be retrieved)?

Reply:
Yes, have confirmed this